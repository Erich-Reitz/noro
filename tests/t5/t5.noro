// standard library functions
function writestrnewline(string any) -> int {}
function writestr(string any) -> int {}
function writeint(int any) -> int {}

function main() -> int {
    int five = 5; 
    int ten = 10; 

    int sumOfFiveAndTen = five + ten;
    if (sumOfFiveAndTen == 20) {
        writestrnewline("never"); 
    } else {
        writestrnewline("always1"); 
        if (sumOfFiveAndTen == 15) {
            writestrnewline("always2"); 
            int one = 1; 
            int two = 2; 
            // mutating 10 in nested scope
            ten = 9; 
            if (one + two == 3) {
                writestrnewline("always3"); 
            } else {
                writestrnewline("never3"); 
            }
        } else {
            writestrnewline("never2"); 
        }
        writestrnewline("always4");
    }
    writestrnewline("always5");
    return ten;
}
