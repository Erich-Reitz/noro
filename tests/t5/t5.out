always1
always2
always3
always4
always5
